src/common.ts

```diff
@@ -152,7 +159,92 @@ export const unpackSDJWT: UnpackSDJWT = (sdJWT, disclosures) => {
   const hash_alg = getHashAlgorithm(sdJWT);
   const map = createHashMapping(disclosures, hash_alg);
 
-  // remove SD_HASH_ALG property
   delete sdJWT[SD_HASH_ALG];
   return unpack({ obj: sdJWT, map });
 };
+
+const createDisclosure = async (
+  claim: DisclosureClaim,
+  hasher: Hasher,
+  options?: {
+    generateSalt?: SaltGenerator;
+  },
+): Promise<{
+  hash: string;
+  disclosure: string;
+}> => {
+  let disclosureArray;
+  const saltGenerator = options?.generateSalt ? options.generateSalt : generateSalt;
+  const salt = saltGenerator(16);
+  if (claim.key) {
+    disclosureArray = [salt, claim.key, claim.value];
+  } else {
+    disclosureArray = [salt, claim.value];
+  }
+
+  const disclosure = base64url.encode(JSON.stringify(disclosureArray));
+  const hash = await hasher(disclosure);
+  return {
+    hash,
+    disclosure,
+  };
+};
+
+/**
+ * Creates a SD-JWT from claims and disclosureFrame definition
+ *
+ * @param claims
+ * @param disclosureFrame declares which properties to be selectively disclosable
+ * @param hasher
+ * @returns
+ */
+export const packSDJWT: PackSDJWT = async (claims, disclosureFrame, hasher, options) => {
```

Comment:
@ragnika what's the plan for `_sd_alg`? Even if that is `sha-256`, shouldn't we include it?

Reply:
@linasi good point, I didn't think of that one thoroughly enough